### ISSUE TITLE
Message "Images in different <source> elements shouldn’t be the same" incorrectly reported for srcset on source element

### ISSUE THREAD

**Issue by AMA3** [ausi/respimagelint#17]
[index.html.txt](https://github.com/ausi/respimagelint/files/300851/index.html.txt)

Using two source elements, one for portrait mode and the other for landscape. Linter reports "Images in different <source> elements shouldn’t be the same" when in fact it is different images in the second srcset attribute (not different source elements) that are the same, for different resolutions.  See the attached index.html and sample images.
![photo500](https://cloud.githubusercontent.com/assets/1852728/15829569/53ce51dc-2be2-11e6-983e-8b68593a9fad.jpg)
![photo1000](https://cloud.githubusercontent.com/assets/1852728/15829572/53d1d744-2be2-11e6-9d5e-02b1871d1931.jpg)
![photo2000](https://cloud.githubusercontent.com/assets/1852728/15829573/53d223de-2be2-11e6-9e48-139fcde58bfd.jpg)

![photo250](https://cloud.githubusercontent.com/assets/1852728/15829571/53d141e4-2be2-11e6-8977-47b70d0aa374.jpg)
![photo440](https://cloud.githubusercontent.com/assets/1852728/15829570/53cff79e-2be2-11e6-9563-1c2ee0709c63.jpg)

**Comment by AMA3** [comment]
```
<!DOCTYPE html>
<html lang=en-US>
  <head>
    <meta charset=utf-8>
    <title>Responsive Linter Bug Demo</title>
    <meta name=viewport content="width=device-width,initial-scale=1">
    <style>body,div,picture,img{width:100%}</style>
    </head>
  <body>
    <div>
      <picture>
        <source
          media="(orientation:portrait) and (max-width:440px)" 
          sizes=100vw
          srcset="photo440.jpg 440w">
        <source
          media="(orientation:landscape) or (min-width:441px)" 
          sizes=100vw
          srcset="photo2000.jpg 2000w, photo1000.jpg 1000w, photo500.jpg 500w, photo250.jpg 250w">
        <img src="photo1000.jpg" alt="example">
        </picture>
      </div>
    </body>
  </html>

```

**Comment by ausi** [comment]
`(orientation:landscape) or (min-width:441px)` is not a valid media query, the correct form would be `(orientation:landscape), (min-width:441px)`.

But since the second `<source>` should always match if the first one doesn’t you should remove the `media` attribute from it:

```html
<picture>
	<source
		media="(orientation:portrait) and (max-width:440px)"
		sizes=100vw
		srcset="photo440.jpg 440w"
	>
	<source
		sizes=100vw
		srcset="photo2000.jpg 2000w, photo1000.jpg 1000w, photo500.jpg 500w, photo250.jpg 250w"
	>
	<img src="photo1000.jpg" alt="example">
</picture>
```

or move the attribute to the `<img>`:

```html
<picture>
	<source
		media="(orientation:portrait) and (max-width:440px)"
		sizes=100vw
		srcset="photo440.jpg 440w"
	>
	<img
		src="photo1000.jpg"
		alt="example"
		sizes=100vw
		srcset="photo2000.jpg 2000w, photo1000.jpg 1000w, photo500.jpg 500w, photo250.jpg 250w"
	>
</picture>
```

**Comment by AMA3** [comment]
Beautiful, thank you!
